### PR TITLE
Uniform arrays

### PR DESCRIPTION
**Pull request by pjcozzi**
For consistency with the uniform changes in #2490.  This is a pull request into that branch.

It kinda looks like a performance win, but if we took more data I'm pretty sure we would see that it is just a breakeven (I didn't expect it to be faster).

Everest fullscreen-ish Retina, % total `_setUniforms`:

```
uniforms-2: 11.71, 13.92, 12.56
uniform-arrays-2: 12.05, 13.28, 10.94
```

### CODE REVIEW

File: Source/Renderer/getUniformArray.js [hunk 1]
```diff
@@ -0,0 +1,575 @@
...
+        Matrix3,
+        Matrix4,
+        RuntimeError) {
+    "use strict";
+    /*global console*/
+
+    /**
+     * @private
+     */
+    var getUniformArray = function(gl, activeUniform, uniformName, locations) {
```
Comment (mramato) [inline Source/Renderer/getUniformArray.js]: would `createUniformArray` be a better name for this function?  `get` usually indicates the thing you're getting already exists.

**Comment by mramato** [comment]
Just the one comment.

**Comment by bagnell** [comment]
Both PRs look good to me. @mramato's comment applies to `getUniform` as well. He can merge when he's happy.

**Comment by pjcozzi** [comment]
Renamed.  This and #2490 are ready.

**Comment by pjcozzi** [comment]
I think this was merged in the wrong order.  It didn't get into master.

**Comment by pjcozzi** [comment]
It is just in `uniforms-2`.

**Comment by mramato** [comment]
My bad, I'll fix it.

**Comment by mramato** [comment]
Okay, we should be good to go.  Sorry again.

**Pull request merged**